search/includes/classes/class-queue.php

```diff
@@ -472,9 +484,11 @@ public function intercept_ep_sync_manager_indexing( $bail, $sync_manager, $index
 			$this->queue_object( $object_id, $indexable_slug );
 		}
 
-		// Queue up a cron event to process these immediately
-		$this->cron->schedule_batch_job();
-
+		// If indexing operations are NOT currently ratelimited, queue up a cron event to process these immediately.
+		if ( false === wp_cache_get( self::INDEX_QUEUEING_ENABLED_KEY, self::INDEX_COUNT_CACHE_GROUP ) ) {
```

Comment:
If the cache entry doesn't exist, `wp_cache_result()` will return `null`, which will not pass this conditional. What do you think about a simple helper function to check rate limit status? So that we're not duplicating the "is rate limited" logic in N spots. `Queue::is_rate_limited()`

Reply:
@nickdaugherty 

wp_cache_get returns false if the cache entry doesn't exist in both the [docs](https://developer.wordpress.org/reference/functions/wp_cache_get/#return) and our [wp-memcached implementation](https://github.com/Automattic/wp-memcached/blob/master/object-cache.php#L283-L331).

It makes sense to move it to a function for readability though and to handle the possibility of future changes to index rate limiting detection.

Reply:
Are you sure?

```
wp> $result = wp_cache_get( 'nothingblahblahblah' );
=> bool(false)
wp> wp> var_dump( $result );
bool(false)
=> NULL
```

Reply:
Oh damnit, I misread the output, you're right 😄

Reply:
@nickdaugherty Pretty sure.I think wp is just being weird:

```
wp> var_dump( false );
bool(false)
=> NULL
```

Reply:
Yeah it prints the result of the line also, I mixed up the stdout from `var_dump()` and that part, need more coffee...